sdcm/nemesis.py

```diff
@@ -1088,6 +1089,68 @@ def disrupt_ldap_connection_toggle(self):
         ContainerManager.unpause_container(self.tester.localhost, 'ldap')
         self.log.info('finished with nemesis')
 
+    def disrupt_ldap_role_add_remove_permission(self):
+        """
+        Scenario:
+        1. Create new user in Scylla
+        2. Create new role in Ldap containing the new user as a member.
+        3. Verify this new user is unauthorised on resources (create a keyspace).
+        4. Create a new super-user role in Scylla.
+        5. Create a new super-user role in Ldap, with the new user as a member.
+        6. Verify this new user is authorised on resources.
+        7. Modify Ldap role to remove membership of new user.
+        8. Verify the new user is unauthorised on resources (create a table).
+        9. Do clean-up: delete Ldap Role, Scylla roles and keyspace/tables.
+
+        """
+        if not self.cluster.params.get('use_ldap_authorization'):
+            raise UnsupportedNemesis('Cluster is not configured to run with LDAP authorization, hence skipping')
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis('Cluster is not enterprise. LDAP is supported only for enterprise. Skipping')
+        node = self.cluster.nodes[0]
+        superuser_role = 'superuser_role'
+        new_test_user = 'new_test_user'
+        self.log.debug("Create new user in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=new_test_user, is_superuser=False,
+                                          is_login=True)
+        with pytest.raises(Unauthorized, match="has no CREATE permission"):
+            with self.cluster.cql_connection_patient(node=node, user=new_test_user, password=LDAP_PASSWORD) as session:
+                session.execute(
+                    """ CREATE KEYSPACE IF NOT EXISTS customer WITH replication = {'class': 'SimpleStrategy',
+                    'replication_factor': 1} """)
+
+        self.log.debug("Create a new super-user role in Scylla")
+        self.tester.create_role_in_scylla(node=node, role_name=superuser_role, is_superuser=True,
+                                          is_login=False)
+        self.cluster.wait_for_schema_agreement()
```

Comment:
shouldn't you change it to `wait_for_user_roles_update` ?

Reply:
no , this one is after only scylla roles creation.
the `wait_for_user_roles_update` comes just ahead, after Ldap role changes (that's something the test verifies),